### ISSUE TITLE
A problem in determining the rotor type with hessian.thermo

### ISSUE THREAD

**Issue by sunchong137** [pyscf/pyscf#2401]
Hi there,

I encountered several problems when using functions in `pyscf.hessian.thermo` for the HCl molecule. I found that it was because `_get_rotor_type()` decided HCl is `REGULAR`, not `LINEAR`.  

In the `_get_rotor_type()` function, the condition for the molecule to be `LINEAR` is `rot_const[0] > 1e8 and (rot_const[1] - rot_const[2] < 1e-3)`. However, the `rot_const[0]` for my HCl molecule was -4e18.

Therefore, my question is, should there be an absolute value function on `rot_const[0]` in the above condition? 

Thanks,
Chong

**Comment by tberkel** [comment]
Yes, this looks like a division by numerical zero issue. Probably better is that `rotational_const()` should return the absolute value of `e`. The principal moments of inertia should all be positive (or exactly zero).

**Comment by sunchong137** [comment]
I agree. The three constants should be non-negative. 
Thanks, Tim!

**Comment by tberkel** [comment]
I'll leave this open for a few days in case anyone else has other suggestions. After that, I can submit a bugfix PR.
